subsys/usb/usb_device.c

```diff
@@ -1226,9 +1226,7 @@ static int custom_handler(struct usb_setup_packet *pSetup,
 			continue;
 		}
 
-		if ((iface->custom_handler) &&
-		    (if_descr->bInterfaceNumber ==
```

Comment:
Please check that all custom handlers behave correctly if the interface number is not intended for it.

Reply:
I've checked current status of class implementation in terms of using *custom_handler*.
3 classes require attention and I would like to speak about it sync meeting.

- [ ] [hid class](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/usb/class/hid/core.c#L491) There is no problem with limitation reduction. Class custom handler checks if the requested interface number matches registered hid devices.

- [ ] [dfu class](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/usb/class/usb_dfu.c#L657) Similar solution as hid has could be implemented but the question is if this is relevant? Shouldn't dfu be single-function USB device? I'd like to discuss.
- [ ] [web usb](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/subsys/usb/webusb/src/webusb.c#L86) Same concerns as above + this is implemented as sample, not as class.